app/components/primer/octicon_component.rb

```diff
@@ -27,20 +29,25 @@ class OcticonComponent < Primer::Component
     # @param size [Symbol] <%= one_of(Primer::OcticonComponent::SIZE_MAPPINGS) %>
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
     def initialize(icon_name = nil, icon: nil, size: SIZE_DEFAULT, **system_arguments)
-      @icon = icon_name || icon
       @system_arguments = system_arguments
-
-      @system_arguments[:class] = Primer::Classify.call(**@system_arguments)[:class]
-      @system_arguments[:height] ||= SIZE_MAPPINGS[size]
+      @system_arguments[:tag] = :svg
 
       # Filter out classify options to prevent them from becoming invalid html attributes.
       # Note height and width are both classify options and valid html attributes.
-      octicon_helper_options = @system_arguments.slice(:height, :width)
-      @system_arguments = add_test_selector(@system_arguments).except(*Primer::Classify::VALID_KEYS, :classes).merge(octicon_helper_options)
+      octicon_options = {
+        height: SIZE_MAPPINGS[fetch_or_fallback(SIZE_OPTIONS, size, SIZE_DEFAULT)]
+      }.merge(@system_arguments.slice(:height, :width, :class))
+
+      @icon = Octicons::Octicon.new(icon_name || icon, octicon_options)
+
+      @system_arguments[:classes] = [@icon.options[:class], @system_arguments[:classes]].compact.join(" ")
+      @system_arguments.merge!(@icon.options.reject { |k| [:class].include?(k) })
     end
 
     def call
-      octicon(@icon, { **@system_arguments })
+      # rubocop:disable Rails/OutputSafety
+      render(Primer::BaseComponent.new(**@system_arguments)) { @icon.path.html_safe }
```

Comment:
I'm not sure if there's a preferred way to handle this `html_safe` here.

Reply:
I think this is fine 👍🏻